src/player.cpp

```diff
@@ -881,13 +881,19 @@ void Player::sendPing()
 		setAttackedCreature(nullptr);
 	}
 
-	if (noPongTime >= 60000 && canLogout()) {
-		if (g_creatureEvents->playerLogout(this)) {
-			if (client) {
-				client->logout(true, true);
-			} else {
-				g_game.removeCreature(this, true);
-			}
+	if (noPongTime >= (pzLocked ? 60000 : 60000 + vocation->getNoPongKickTimeChange()))) {
```

Comment:
Well, I guess it should be std::max here, since someone can increase the kick time for given vocation 🤔

And perhaps no pong kick with pz lock should be configurable.